fdbserver/DataDistribution.actor.cpp

```diff
@@ -784,7 +787,11 @@ struct DDTeamCollection : ReferenceCounted<DDTeamCollection> {
 			if( req.wantsTrueBest ) {
 				ASSERT( !bestOption.present() );
 				for( int i = 0; i < self->teams.size(); i++ ) {
-					if( self->teams[i]->isHealthy() && (!req.preferLowerUtilization || self->teams[i]->hasHealthyFreeSpace()) ) {
+					if (self->teams[i]->isHealthy() &&
+					    (!req.preferLowerUtilization || self->teams[i]->hasHealthyFreeSpace()) &&
+					    (!req.requiresAssignedData || self->shardsAffectedByTeamFailure->getShardsFor(ShardsAffectedByTeamFailure::Team(self->teams[i]->getServerIDs(), self->primary)) .size() > 0) &&
+					    self->teams[i]->getMinFreeSpaceRatio() >= req.minFreeSpaceRatio) 
```

Comment:
This ` teamList[j]->getMinFreeSpaceRatio() >= req.minFreeSpaceRatio` condition will cause SS stops to balance once SS is below the CUTOFF value.
If a cluster has many such SSes, we will lose DD's ability for data balance and some SS may fill up very quickly.

In addition, this will also make `FREE_SPACE_RATIO_DD_CUTOFF` less meaningful (if not meaningless). 

IMO, this condition can be turned on only when the average SS utilization is low.

Reply:
This is the same behavior as before, so I’m hesitant to change it in this PR.

Reply:
My bad. I misread this to something else (while I'm in a sleepy mode). :(